examples/cluster/plot_adjusted_for_chance_measures.py

```diff
@@ -3,135 +3,229 @@
+#   cluster assignment has an ARI of 0.0 in expectation;
+#
+# - Mutual Information (MI) is an information theoretic measure that quantifies
+#   how dependent are the two labelings. Note that the maximum value of MI for
+#   perfect labelings depends on the number of clusters and samples;
+#
+# - Normalized Mutual Information (NMI), a Mutual Information defined between 0
+#   (no mutual information) in the limit of large number of data points and 1
+#   (perfectly matching label assignments, up to a permutation of the labels).
+#   It is not adjusted for chance: then the number of clustered data points is
+#   not large enough, the expected values of MI or NMI for random labelings can
+#   be significantly non-zero;
+#
+# - Adjusted Mutual Information (AMI), a chance-adjusted Mutual Information.
+#   Similarly to ARI, random cluster assignment has an AMI of 0.0 in
+#   expectation.
+#
+# For more information, see the :ref:`clustering_evaluation` module.
+
 from sklearn import metrics
 
+score_funcs = [
+    ("V-measure", metrics.v_measure_score),
+    ("Rand index", metrics.rand_score),
+    ("ARI", metrics.adjusted_rand_score),
+    ("MI", metrics.mutual_info_score),
+    ("NMI", metrics.normalized_mutual_info_score),
+    ("AMI", metrics.adjusted_mutual_info_score),
+]
 
-def uniform_labelings_scores(
-    score_func, n_samples, n_clusters_range, fixed_n_classes=None, n_runs=5, seed=42
-):
-    """Compute score for 2 random uniform cluster labelings.
+# %%
+# First experiment: fixed ground truth labels and growing number of clusters
+# --------------------------------------------------------------------------
+#
+# We first define a function that creates uniformly-distributed random labeling.
 
-    Both random labelings have the same number of clusters for each value
-    possible value in ``n_clusters_range``.
+import numpy as np
 
-    When fixed_n_classes is not None the first labeling is considered a ground
-    truth class assignment with fixed number of classes.
-    """
-    random_labels = np.random.RandomState(seed).randint
-    scores = np.zeros((len(n_clusters_range), n_runs))
 
-    if fixed_n_classes is not None:
-        labels_a = random_labels(low=0, high=fixed_n_classes, size=n_samples)
+def random_labels(n_samples, n_classes):
+    labels = np.random.randint
+    return labels(low=0, high=n_classes, size=n_samples)
 
-    for i, k in enumerate(n_clusters_range):
-        for j in range(n_runs):
-            if fixed_n_classes is None:
-                labels_a = random_labels(low=0, high=k, size=n_samples)
-            labels_b = random_labels(low=0, high=k, size=n_samples)
-            scores[i, j] = score_func(labels_a, labels_b)
-    return scores
 
+# %%
+# Another function will use the `random_labels` function to create a fixed set
+# of ground truth labels (`labels_a`) distributed in `n_classes` and then score
+# several sets of randomly "predicted" labels (`labels_b`) to assez the
```

Comment:
I'm not sure if assez makes sense here: https://en.wiktionary.org/wiki/assez. Did you mean [assess](https://www.merriam-webster.com/dictionary/assess)?

Reply:
Good catch! I guess my brain was in french mode. Thanks for your comments :)